core/src/main/java/com/linecorp/armeria/server/docs/ServiceSpecification.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.docs;
+
+import static com.google.common.collect.ImmutableSortedMap.toImmutableSortedMap;
+import static java.util.Objects.requireNonNull;
+
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.function.Function;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.Streams;
+
+import com.linecorp.armeria.server.Service;
+
+/**
+ * The specification of one or more {@link Service}s that provides their {@link ServiceInfo}s and
+ * {@link ClassInfo}s.
+ */
+public final class ServiceSpecification {
+
+    /**
+     * Merges the specified {@link ServiceSpecification}s into one.
+     */
+    public static ServiceSpecification merge(Iterable<ServiceSpecification> specs) {
+        return new ServiceSpecification(
+                Streams.stream(specs).flatMap(s -> s.services().values().stream()).iterator(),
```

Comment:
I don't know if it's necessarily better but I think you'd be able to remove the private constructor if you used ```::iterator``` here to convert the stream to an iterable